src/storage/data_table.cpp

```diff
@@ -437,7 +437,57 @@ static void VerifyCheckConstraint(TableCatalogEntry &table, Expression &expr, Da
 	}
 }
 
-void DataTable::VerifyAppendConstraints(TableCatalogEntry &table, DataChunk &chunk) {
+static TableIndexList &GetForeignKeyConstraintIndicesAndChunk(const string &table_name, const vector<idx_t> &src_keys,
+                                                              const vector<idx_t> &dst_keys, ClientContext &context,
+                                                              DataChunk &src_chunk, DataChunk &dst_chunk) {
+	auto table_entry_ptr =
+	    Catalog::GetCatalog(context).GetEntry<TableCatalogEntry>(context, INVALID_SCHEMA, table_name);
+	if (table_entry_ptr == nullptr) {
+		throw ParserException("Can't find table \"%s\" in foreign key constraint", table_name);
+	}
+	TableIndexList &indices = table_entry_ptr->storage->info->indexes;
+
+	// make the data chunk to check
+	vector<LogicalType> types;
+	for (idx_t i = 0; i < table_entry_ptr->columns.size(); i++) {
+		types.emplace_back(table_entry_ptr->columns[i].type);
+	}
+	dst_chunk.InitializeEmpty(types);
```

Comment:
This is equivalent to `CreateMockChunk`, could you use that instead?

Reply:
It is not equivalent to CreateMockChunk. The difference is the mapping from src_keys to dst_keys for same index.

                 dst_chunk.data[dst_keys[i]].Reference(src_chunk.data[src_keys[i]]);

Reply:
Ah correct, there is a subtle difference indeed.